### DISCUSSION TITLE
AG Grid Export to CSV in Native Mode

### DISCUSSION THREAD

**Discussion by LeoM1970** [zauberzeug/nicegui#3002] [Q&A] [answered]
### Question

I am enjoying your project, as a novice to GUI creation.

I would like to save the contents of an editable AG Grid.

I followed your advice from https://www.reddit.com/r/nicegui/comments/11q434m/how_can_i_implement_ag_grids_export_as_csv_feature/?rdt=58795, and noting from the documentation that **call_api_method** is deprecated, I have used **run_grid_method** instead.

This works well but as I intend to package the UI via PyInstaller, I then tried it in Native Mode which does not appear to work. 

- **Is there an alternative method to save the AG Grid in Native Mode?**

Here is an MRE using a grid borrowed from your documentation:

```py
from nicegui import ui

grid = ui.aggrid({
    'defaultColDef': {'flex': 1},
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
        {'headerName': 'Parent', 'field': 'parent', 'hide': True},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18, 'parent': 'David'},
        {'name': 'Bob', 'age': 21, 'parent': 'Eve'},
        {'name': 'Carol', 'age': 42, 'parent': 'Frank'},
    ],
    'rowSelection': 'multiple',
}).classes('max-h-40')

ui.button('Export', on_click=lambda: grid.run_grid_method('exportDataAsCsv'))

ui.run(native=True)  # export button above only works when removing native=True
```

**Comment by falkoschindler** [comment, marked as answer]
Hi @LeoM1970,

We just released a new feature allowing to configure pywebview settings to allow downloads: https://github.com/zauberzeug/nicegui/pull/2884#issuecomment-2058882185.
Does that work for you?

**Comment by LeoM1970** [comment]
Thank you, @falkoschindler 

The MRE updated as below does work for me now.

```py
from nicegui import ui, app

app.native.settings['ALLOW_DOWNLOADS'] = True

grid = ui.aggrid({
    'defaultColDef': {'flex': 1},
    'columnDefs': [
        {'headerName': 'Name', 'field': 'name'},
        {'headerName': 'Age', 'field': 'age'},
        {'headerName': 'Parent', 'field': 'parent', 'hide': True},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 18, 'parent': 'David'},
        {'name': 'Bob', 'age': 21, 'parent': 'Eve'},
        {'name': 'Carol', 'age': 42, 'parent': 'Frank'},
    ],
    'rowSelection': 'multiple',
}).classes('max-h-40')

ui.button('Export', on_click=lambda: grid.run_grid_method('exportDataAsCsv'))

ui.run(native=True)

```
